src/agent/payload.cc

```diff
@@ -103,37 +104,43 @@ void PointCallHandle::SetReturnValue(long ret_val) {
 void
 wrapper_entry(sp::SpPoint* pt,
               sp::PayloadFuncEntry entry) {
-  sp_debug("In wrapper entry function for point %p", pt);
-  if (sp::g_context == NULL) {
-    sp_perror("Global context is NULL, return");
-    return;
-  }
 
-  void* user_info = NULL;
-  sp::SpFunction* callee = sp::Callee(pt);
+  if (trampGuard == 0){
+    trampGuard = 1;	  
+    sp_debug("In wrapper entry function for point %p", pt);
+    
+    if (sp::g_context == NULL) {
+      sp_perror("Global context is NULL, return");
```

Comment:
Here the trampGuard is left as 1 if this path is taken and probably not correct.